installer/datafiles/base_omsagent.data

```diff
@@ -191,13 +191,33 @@ fi
 #   6. Run the %postun hook of the old package.
 #
 # Thus, if we're an upgrade, skip all of this cleanup
+ETC_DIR=/etc/opt/microsoft/omsagent
+WORKSPACE_REGEX='^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$'
 if ${{PERFORMING_UPGRADE_NOT}}; then
-   # Clean up directory tree (created via PostInstall) if dirs are empty
+   # Clean up /var directory tree
    rm -rf /var/opt/microsoft/omsagent 2> /dev/null
-
+   
+   # Clean up /etc/opt directory tree (created via PostInstall) if dirs are empty
+   # If multi-workspace scenario
+   for ws_id in `ls -1 $ETC_DIR | grep -E $WORKSPACE_REGEX`
+    do
+        echo "Deleteting certs and conf directories for workspace $ws_id if empty..."
+        rmdir /etc/opt/microsoft/omsagent/$ws_id/certs/ 2> /dev/null
+        rmdir /etc/opt/microsoft/omsagent/$ws_id/conf/ 2> /dev/null       
+    done
+   
    # Clean up installinfo.txt file (registered as "conf" file to pass rpmcheck)
    rm -f /etc/opt/microsoft/omsagent/sysconf/installinfo.txt*
-   rm -rf /etc/opt/microsoft/omsagent 2> /dev/null
+   rmdir /etc/opt/microsoft/omsagent/sysconf 2> /dev/null
+   # Clean up symbolic links if multi-workspace scenario
+   rm /etc/opt/microsoft/omsagent/certs/ 2> /dev/null
+   rm /etc/opt/microsoft/omsagent/conf/ 2> /dev/null
+   
+   # Clean up directory tree if agent installed or upgraded without onboarding. No symbolic links.
+   rmdir /etc/opt/microsoft/omsagent/certs/ 2> /dev/null
+   rmdir /etc/opt/microsoft/omsagent/conf/ 2> /dev/null
+   # Remove parent folders
+   rmdir /etc/opt/microsoft/omsagent 2> /dev/null
    rmdir /etc/opt/microsoft 2> /dev/null
```

Comment:
If this line still runs, where is the code that prevents any customer's custom config (i.e. `/etc/opt/microsoft/omsagent/<workspace id>/conf/custom.conf`) from getting deleted? Won't the entire folder still get deleted?